lightning/src/ln/channel.rs

```diff
@@ -1717,8 +1765,28 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 				removed_outbound_total_msat += htlc.amount_msat;
 			}
 		}
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::<ChanSigner>::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 + removed_outbound_total_msat {
-			return Err(ChannelError::Close("Remote HTLC add would put them over their reserve value"));
+
+		if self.channel_outbound {
+			// Check that they won't violate our channel reserve by adding this HTLC.
+
+			// One of the +1's is for the HTLC that is currently being added to the commitment tx
+			// and the other is a fee spike buffer we're keeping for the remote (this deviates
+			// from the spec but should help protect us from stuck channels).
+			// TODO: is the reason this added HTLC's amount can't count toward's the receiver's
+			// fee spike buffer because that doesn't work with existing HTLC output spend scripts?
+			let local_fee_cost_msat = self.commit_tx_fee_msat(self.htlc_count_next_local_commit_tx() + 1 + 1);
+			if self.value_to_self_msat < self.their_channel_reserve_satoshis * 1000 + local_fee_cost_msat {
+				return Err(ChannelError::Ignore("Cannot receive value that would put us over their reserve value"));
```

Comment:
This applies to a few places, but returning an ::Ignore is gonna end up getting the channel closed (::Ignore means "ignore this message", but that implies that the signature in their next commitment_signed is gonna be "wrong"). What we need to do is swap the pending_forward_state out for a ::Fail version so that we auto-fail the HTLC after we accept it.

Reply:
Gotcha, `temporary_channel_failure` seems like the only error that fits

Reply:
Hm. For a `temporary_channel_failure`, I need to be able to create a channel update within a `Channel`. 

This means adding two extra fields to the `Channel` struct afaict: a FeeEstimator and what network the channel is on. I'm hesitant to add more fields to the Channel struct so lmk if there's an alternative you can think of.